persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/slick/SlickReadSideImpl.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.scaladsl.persistence.slick
+
+import akka.{ Done, NotUsed }
+import akka.persistence.query.Offset
+import akka.stream.scaladsl.Flow
+import com.lightbend.lagom.internal.persistence.jdbc.{ SlickOffsetDao, SlickOffsetStore, SlickProvider }
+import com.lightbend.lagom.scaladsl.persistence.ReadSideProcessor.ReadSideHandler
+import com.lightbend.lagom.scaladsl.persistence.slick.SlickReadSide
+import com.lightbend.lagom.scaladsl.persistence.{ AggregateEvent, AggregateEventTag, EventStreamElement }
+import org.slf4j.LoggerFactory
+import slick.dbio.{ DBIOAction, NoStream }
+import slick.jdbc.JdbcProfile
+import slick.jdbc.JdbcBackend.Database
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.reflect.ClassTag
+
+class SlickReadSideImpl(slick: SlickProvider, offsetStore: SlickOffsetStore)(implicit val executionContext: ExecutionContext)
+  extends SlickReadSide {
+
+  override val db: Database = slick.db
+  override val profile: JdbcProfile = slick.profile
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  override def builder[Event <: AggregateEvent[Event]](readSideId: String): ReadSideHandlerBuilder[Event] = new ReadSideHandlerBuilder[Event] {
+    var globalPrepare: DBIOAction[Any, _, _] = DBIOAction.successful()
+    var prepare: (AggregateEventTag[Event]) => DBIOAction[Any, NoStream, Nothing] = (_) => DBIOAction.successful()
```

Comment:
`DBIOAction.successful()` in lines 30 and 31 introduced two compilation warnings:

```

[warn] /Users/ignasi/git/projects/lightbend/lagom/persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/slick/SlickReadSideImpl.scala:31: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
[warn]         signature: DBIOAction.successful[R](v: R): slick.dbio.DBIOAction[R,slick.dbio.NoStream,slick.dbio.Effect]
[warn]   given arguments: <none>
[warn]  after adaptation: DBIOAction.successful((): Unit)
[warn]     var prepare: (AggregateEventTag[Event]) => DBIOAction[Any, NoStream, Nothing] = (_) => DBIOAction.successful()
[warn]                                                                                                                 ^
```

TIL, `Automatic Unit inference has been deprecated in scala 2.11`(see [SO](http://stackoverflow.com/questions/25351988/adaptation-of-argument-list-by-inserting-has-been-deprecated) ) which means we should explicitly pass `Unit` ( or `()`) as an argument when we want to type to `Unit`. Then `DBIOAction.successful()` should become `DBIOAction.successful(())`.

As we move close to migrating to 2.12 I prefer keeping the warning count as it is. ;-)